lib/bundler/source_list.rb

```diff
@@ -77,12 +77,10 @@ def get(source)
     end
 
     def lock_sources
+      lock_sources = (path_sources + git_sources + plugin_sources).sort_by(&:to_s)
       if Bundler.feature_flag.lockfile_uses_separate_rubygems_sources?
-        [[default_source], @rubygems_sources, git_sources, path_sources, plugin_sources].map do |sources|
-          sources.sort_by(&:to_s)
-        end.flatten(1)
+        lock_sources + rubygems_sources.sort_by(&:to_s)
```

Comment:
Here I'm reusing the `rubygems_sources` in this file, which is defined as

```
    def rubygems_sources
      @rubygems_sources + [default_source]
    end
```

Maybe I should undo that change to make this diff more clear.

Reply:
So, it's the same 5 components that existed previously (`[[default_source], @rubygems_sources, git_sources, path_sources, plugin_sources]`), I'm just changing the order to match the previous order.

The difference from before is still mantained, we have separated rubygems sources since the `combine_rubygems_sources` that merged them into a single section is no longer used.